language/libra-tools/writeset-transaction-generator/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use libra_types::{
+    account_address::AccountAddress,
+    transaction::{Script, Transaction},
+};
+use libra_types::{account_config::libra_root_address, transaction::WriteSetPayload};
+use std::io::Write;
+use stdlib::compile_script;
+use tempfile::NamedTempFile;
+
+fn compile_admin_script(input: &str) -> Result<Script> {
+    let mut temp_file = NamedTempFile::new()?;
+    temp_file.write_all(input.as_bytes())?;
+    let cur_path = temp_file.path().to_str().unwrap().to_owned();
+    Ok(Script::new(compile_script(cur_path), vec![], vec![]))
+}
+
+pub fn encode_remove_validators_transaction(validators: Vec<AccountAddress>) -> Transaction {
```

Comment:
What would you think of storing these scripts in a separate directory e.g., `admin_scripts` and then use something like [mustache](https://github.com/nickel-org/rust-mustache) to templatize these scripts in a similar manner to what we do in the functional tests? Mustache can encode loops, so it could recreate this logic pretty succinctly I believe. 

The command line could then pass the path to the script along with a comma-separated list of arguments that would be picked up substituted in to the script body?